expression/integration_test.go

```diff
@@ -7895,3 +7895,17 @@ func TestIssue39146(t *testing.T) {
 	tk.MustExec("set @@tidb_enable_vectorized_expression = off;")
 	tk.MustQuery(`select str_to_date(substr(dest,1,6),'%H%i%s') from sun;`).Check(testkit.Rows("20:23:10"))
 }
+
+func TestIssue35289(t *testing.T) {
```

Comment:
```TestIssue35289``` is a bad name of test cases. we should avoid it. 

I think it is a corner case about insert and we can put it into ```TestInsert``` and comment the issue id.